Framework/Reflectometry/src/ReflectometryWorkflowBase2.cpp

```diff
@@ -681,37 +722,34 @@ std::string ReflectometryWorkflowBase2::findProcessingInstructions(const Instrum
 
   const std::string analysisMode = getProperty("AnalysisMode");
 
+  boost::optional<size_t> maybeStart;
+  boost::optional<size_t> maybeStop;
   if (analysisMode == "PointDetectorAnalysis") {
-    const std::vector<double> pointStart = instrument->getNumberParameter("PointDetectorStart");
-    const std::vector<double> pointStop = instrument->getNumberParameter("PointDetectorStop");
-
-    if (pointStart.empty() || pointStop.empty())
-      throw std::runtime_error("Could not find 'PointDetectorStart' and/or "
-                               "'PointDetectorStop' in parameter file. Please "
-                               "provide processing instructions manually or "
-                               "set analysis mode to 'MultiDetectorAnalysis'.");
-
-    const auto detStart = static_cast<int>(pointStart[0]);
-    const auto detStop = static_cast<int>(pointStop[0]);
-
-    auto instructions = std::to_string(detStart);
-    if (detStart != detStop)
-      instructions += ":" + std::to_string(detStop);
-    return instructions;
+    maybeStart = getDetectorParamOrNone(instrument, inputWS, "PointDetectorStart");
+    maybeStop = getDetectorParamOrNone(instrument, inputWS, "PointDetectorStop");
+    if (!maybeStart || !maybeStop) {
+      throw std::runtime_error(
+          "Could not find 'PointDetectorStart' and/or 'PointDetectorStop' in parameter file. Please provide processing "
+          "instructions manually or set analysis mode to 'MultiDetectorAnalysis'.");
+    }
+  } else {
+    maybeStart = getDetectorParamOrNone(instrument, inputWS, "MultiDetectorStart");
+    maybeStop = getDetectorParamOrNone(instrument, inputWS, "MultiDetectorStop");
+    if (!maybeStart) {
+      throw std::runtime_error("Could not find 'MultiDetectorStart in parameter file. Please provide processing "
+                               "instructions manually or set analysis mode to 'PointDetectorAnalysis'.");
+    }
+    if (!maybeStop) {
+      // Default to the last workspace index if stop is not given
+      maybeStop = inputWS->getNumberHistograms() - 1;
```

Comment:
Also just want to check that we don't expect `getNumberHistograms()` to return 0?

Reply:
Ah, good point, we don't but I should throw or assert here. I'll have a think about what's best.

Reply:
I've put in an assert for now because we really shouldn't get here with 0 histograms. It should probably be validated earlier on in validateInputs but that's beyond the scope of this PR. I'll create an issue for it.